lib/plugins/aws/deploy/index.js

```diff
@@ -85,7 +87,16 @@ class AwsDeploy {
           if (this.options.noDeploy) {
             return BbPromise.resolve();
           }
-          return this.serverless.pluginManager.spawn('aws:deploy:deploy');
+
+          return BbPromise.bind(this)
+            .then(this.setBucketName)
+            .then(this.shouldBeDeployed)
```

Comment:
Instead of rejecting, the checkForChanges method/event could set a property in the deploy plugin (like this.skipPhysicalDeployments) that will be evaluated by the `uploadArtifacts` and `updateStack` events, so that they only resolve instead of doing any physical uploads or updates.